src/net/server/bytearrstream.rs

```diff
@@ -0,0 +1,452 @@
+use crate::{error::BoxError, ff, net::MpcHelperServerError};
+use async_trait::async_trait;
+use axum::extract::{BodyStream, FromRequest, Query, RequestParts};
+use futures::{ready, Stream};
+use hyper::body::{Bytes, HttpBody};
+use pin_project::pin_project;
+use std::collections::VecDeque;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// the size of the field type specified in the query param of the request
+/// Possible sizes are are currently from:
+/// * `fp2`
+/// * `fp31`
+/// * `fp32_bit_prime`
+struct FieldSize {
+    field_size: u32,
+}
+
+#[cfg(feature = "enable-serde")]
+#[async_trait]
+impl<B: Send> FromRequest<B> for FieldSize {
+    type Rejection = MpcHelperServerError;
+
+    async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
+        #[derive(serde::Deserialize)]
+        struct FieldStr {
```

Comment:
Is it possible to deserialize directly into a string?

Reply:
I believe axum needs the struct in order to know what the key of the query param is